lib/paper_trail/record_trail.rb

```diff
@@ -414,13 +410,29 @@ def add_transaction_id_to(data)
       data[:transaction_id] = PaperTrail.transaction_id
     end
 
+    # Given a HABTM association, returns an array of ids.
+    # @api private
+    def habtm_assoc_ids(habtm_assoc)
+      current = @record.send(habtm_assoc.name).to_a.map(&:id) # TODO: `pluck` would use less memory
```

Comment:
I believe in `ActiveRecord` you're able to call something like `@record.send("#{habtm_assoc.name}_ids") to get the current list of ids, and it should give you the most up to date values for it too...

If the concern is that it might get out of sync you could always do `@record.send("#{habtm_assoc.name}").reload` first.